packages/mui-material/src/Autocomplete/Autocomplete.js

```diff
@@ -592,17 +592,17 @@ const Autocomplete = React.forwardRef(function Autocomplete(inProps, ref) {
           },
         })}
       </AutocompleteRoot>
-      {popupOpen && anchorEl ? (
+      {anchorEl ? (
         <AutocompletePopper
           as={PopperComponent}
           disablePortal={disablePortal}
           style={{
-            width: anchorEl ? anchorEl.clientWidth : null,
+            width: anchorEl && popupOpen ? anchorEl.clientWidth : null,
```

Comment:
This change looks unnecessary to me.

Does it fix something?

Reply:
When the Popper is closed but `keepMounted` is `true`, the Popper is still in the DOM _along with the_ styles. So we don't want to apply the `width` style when Popper is kept mounted. Basically avoid this (see the image below):

![Screenshot (15)](https://user-images.githubusercontent.com/20900032/185329457-0f0d4d9d-87c0-4aad-a455-4b98f5a8b3d8.png).

In the case, when the Popper is closed and keepMounted is not passed to Popper, the DOM itself is removed, so no styles.

Reply:
> We don't want to apply the width style when Popper is kept mounted

Now I got your point but still wonder what's the issue of having the width when Popper is kept mounted (but closed)?

The logic seems to add more complexity without beneficial gain. I suggest removing it if it produces the same result (from the user perspective).

Reply:
@siriwatknp 
> what's the issue of having the width when Popper is kept mounted (but closed)?

There is no issue as such (visually), just didn't want any unnecessary style to be applied.

> The logic seems to add more complexity without beneficial gain. I suggest removing it if it produces the same result (from the user perspective).

Good point. It produces the same result without it. I removed the logic now. Thanks. 👍